evalml/tests/component_tests/test_components.py

```diff
@@ -1067,15 +1075,22 @@ def test_estimator_fit_respects_custom_indices(use_custom_index, estimator_class
 
     supported_problem_types = estimator_class.supported_problem_types
 
-    if ProblemTypes.REGRESSION in supported_problem_types or ProblemTypes.TIME_SERIES_REGRESSION in supported_problem_types:
+    ts_problem = False
+    if ProblemTypes.REGRESSION in supported_problem_types:
         X, y = X_y_regression
+    elif ProblemTypes.TIME_SERIES_REGRESSION in supported_problem_types:
+        X, y = ts_data
+        ts_problem = True
     else:
         X, y = X_y_binary
 
     X = pd.DataFrame(X)
     y = pd.Series(y)
 
-    if use_custom_index:
+    if use_custom_index and ts_problem:
```

Comment:
Rather than needing an extra variable, we could just check if `ProblemTypes.TIME_SERIES_REGRESSION in estimator.supported_problem_types` (don't remember syntax)?